keps/sig-api-machinery/4420-retry-generate-name/README.md

```diff
@@ -0,0 +1,541 @@
+resources for a generateName, say 2M or higher, then retries could become more
+common. 
+
+###### Will enabling / using this feature result in non-negligible increase of resource usage (CPU, RAM, disk, IO, ...) in any components?
+
+No, at least not in all but the most pathologcial of situations.
+
+If a cluster was receiving a high load of create requests for generateName
+resources with a large number (millions) of existing resources already generated
+with the same generateName prefix, this could cause up to a 7x increase in CPU.
+But we do not expect this to actually happen, primarily because etcd cannot
+support these volumes of resources.
+
+###### Can enabling / using this feature result in resource exhaustion of some node resources (PIDs, sockets, inodes, etc.)?
+
+No
+
+### Troubleshooting
+
+###### How does this feature react if the API server and/or etcd is unavailable?
+
+###### What are other known failure modes?
+
+The only know failure modes are pathological. E.g. A runaway system creating
+unbounded resources using generateName could eventually trigger amplified load
+on webhooks with this feature.
+
+###### What steps should be taken if SLOs are not being met to determine the problem?
+
+Look at the number of resources (using kube-state-metrics or similar). If there
+exists a very large number of resources (millions) of a particular resource kind
+and they all share the same generateName prefix, this might be triggering
+high retry load.
+
+## Implementation History
+
+<!--
+Major milestones in the lifecycle of a KEP should be tracked in this section.
+Major milestones might include:
+- the `Summary` and `Motivation` sections being merged, signaling SIG acceptance
+- the `Proposal` section being merged, signaling agreement on a proposed design
+- the date implementation started
+- the first Kubernetes release where an initial version of the KEP was available
+- the version of Kubernetes where the KEP graduated to general availability
+- when the KEP was retired or superseded
+-->
+
+## Drawbacks
+
+This code is in a critical system codepath, if anything is wrong with it, it could cause
+severe problems, up to and including data corruption and loss.
+
+## Alternatives
+
+There are many alternatives that we considered including:
+
+- Increasing the length of the random suffix appended to generateName when generating
+  a name. This would be a breaking change in two ways:
+  - There is a 63 char name length limit. to guarantee a 5 char random suffix
+    length, generateName is trimmed if its length is >58 chars. If the random
+    suffix length were to be increased, generateName would need to trimmed to a
+    shorter length, breaking users that depend on 58 chars of generateName being
+    preserved. (Increasing the max lengh of name would also be a breaking
+    change).
+  - There are major downstream projects that use "{5}" regexes to match the
+    random suffix.
```

Comment:
The 5 character thing is not an API guarantee, so it's an unwelcome change but not breaking in that sense.

Reply:
> so it's an unwelcome change but not breaking in that sense

I agree.  This KEP started with the question:  "Can I stop the bleeding this problem is causing right now without breaking anyone?"  I still seems plausable to me that at some point we'd rip the bandaid off and make the length change, but it will take longer to agree on, implemented, and roll out than this.